patterns/self-signed-vs-user-provided-tls/README.md

```diff
@@ -0,0 +1,107 @@
+# Self-signed Ingress TLS Certificate vs. User-provided
+
+If you are providing a TLS secret with your app to terminate TLS with Kubernetes Ingress, you will likely want the ability to let users decide between providing their own certificate vs. defaulting to a self-signed one.
+
+Source Application: [Mlflow](https://github.com/replicatedhq/platform-examples/blob/main/applications/mlflow)
+
+1. Add the tls configuration to your helm chart values
+
+[Mlflow Helm Values](https://github.com/replicatedhq/platform-examples/blob/main/applications/mlflow/chart/mlflow/values.yaml)
+```yaml
+ingress:
+  enabled: true
+  hostname: chart-example.local
+  tls:
+    enabled: true
+    genSelfSignedCert: true
+    cert: |
+      -----BEGIN CERTIFICATE-----
+      -----END CERTIFICATE-----
+    key: |
+      -----BEGIN PRIVATE KEY-----
+      -----END PRIVATE KEY-----
+```
+
+2. Add a tls secret to your chart and implement the templating to conditionally choose between user-provided and self-signed
```

Comment:
I'm curious, have you addressed certificate rotation/renewal for self signed here?

Reply:
So today this will render a new tls certificate/key each time the helm is rendered. So no need to worry about renewal, but is sort of a different oddity that may not be expected. Would have to use `lookup` to address that. One of the stories I'm planning to create is to provide a cert manager example. that could be our preferred way to recommend addressing TLS

Reply:
I really like cert-manager and that's why I was curious honestly. If we feel like that's the best way to handle it.